docs/design-proposals/arch/daemonset.md

```diff
@@ -48,7 +50,21 @@ This is a variant of the above-mentioned DS in RP: cluster admin manages DaemonS
 
 This architecture mitigates the DS sync problem, but it brings more incompatibility of scale-out and scale-up, as scale-up has no such sync in the first place.
 
-### Reequired Changes
+Regarding the scope of changes to keep track of TP origin, there are options:
+
+**System-tenant Resources Only**
+Only resources of system tenant need to keep track of TP origin; user tenant has not change.
+
+Its pro is smaller memory footprint; smaller impact on existing system design.
+
+**All tenants**
+All reources used in kubelet (and kube-proxy), regardless of tenat they are in, keep track of theit TP origin.
+
+Its pro is consistency, no assumption that user tenant being on one single TP.
+
+We'll implement by keeping track for all tenants.
```

Comment:
+1

Reply:
IIRC, service support also keeps all resource origin. It is a universal design, you can add it as one of the pros.

Reply:
what is the mapping used for ? any other resources it needs to explicitly get the desired TP to get ?

below is a mail we have long time ago:
if just for secrets or configmaps etc. i thought we have looked into this and the kubelet should be able get those info per POD which is under the pod ID.

Secrets, configmaps were stored as paths as below where the POD UID is unique per the tenant-ns-podName. E.g. below is the flannel token secret path on the node.  In a worst case where the system pods were from different TP, it should be OK at the node in terms of loading secrets.
 
./var/lib/kubelet/pods/0a376aa2-9ec3-4fa4-bb48-d85e76991dcc/volumes/kubernetes.io~secret/flannel-token-k65mf

Reply:
1-  will add a sentence stating the discussion applies to resources of both kubelet and kube-proxy (which includes service).
2-  defining one set of the secrets (and possible other supporting resources) that applies to all TPs is interesting; however, it is not the concern of current design.
3-  will rewrite section of requried changes, to cites resources of kubelet to be pod, configmap, secret, service, resourcces of kube-proxy to be service and endpoint.